### DISCUSSION TITLE
Turbopack bundles a `.ts` file as a static media, is it a bug?

### DISCUSSION THREAD

**Discussion by yf-yang** [vercel/next.js#59729] [Help]
### Summary

Reproduction: https://github.com/yf-yang/turbopack-webworker-bug
I am creating a single button in the page, when clicking the button, a new web worker will start. [Here](https://github.com/yf-yang/turbopack-webworker-bug/blob/main/app/WorkerButton.tsx#L10) is the code:

``` typescript
worker = new Worker(new URL("./workerScript.ts", import.meta.url));
```

If I start the site with `pnpm dev` (which uses webpack), things work as expected. When I switch to `pnpm dev --turbo` (which uses turbopack), in the `.next` folder I can see that it moves that `workerScript.ts` to `/_next/static/media/workerScript.ts`, which means the `.ts` is treated as the video extension instead of a TypeScript file. 

Do I miss some additional operation to make it work in turbopack, or it is a bug?

### Example

https://github.com/yf-yang/turbopack-webworker-bug

**Comment by yf-yang** [comment]
Trying to switch to `.js` and `.tsx` and got `Cannot use import statement outside a module`, maybe this is not a supported feature?

**Reply by yf-yang** [reply]
Samilar issue: https://github.com/vercel/turbo/issues/3643

**Comment by asimaranov** [comment]
Have you fixed this? Facing the same issue
